src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -138,6 +145,18 @@ BlockIO InterpreterCreateQuery::createDatabase(ASTCreateQuery & create)
             throw Exception(ErrorCodes::DATABASE_ALREADY_EXISTS, "Database {} already exists.", database_name);
     }
 
+    if (auto max_db = getContext()->getGlobalContext()->getServerSettings().max_database_num_to_throw; max_db > 0)
+    {
+        size_t db_count = DatabaseCatalog::instance().getDatabases().size();
+        // there's an invisible system database _temporary_and_external_tables, so we need to subtract 1
+        if (db_count > 0)
```

Comment:
I think it makes sense to check that this hidden system DB is actually present  `&& DatabaseCatalog::instance().isDatabaseExist(DatabaseCatalog::TEMPORARY_DATABASE)`